bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Region.java

```diff
@@ -616,7 +629,158 @@ public void translate (Point pt) {
 @Override
 public String toString () {
 	if (isDisposed()) return "Region {*DISPOSED*}";
-	return "Region {" + handle + "}";
+	return "Region " + zoomToHandle;
+}
+
+class Operation {
+	enum OperationType {
+		ADD, SUBTRACT, INTERSECT, TRANSLATE
+	}
+	enum ArgumentType {
+		REGION, RECTANGLE, POINTS, POINT
+	}
+
+	private OperationType operationType;
+	private ArgumentType argumentType;
+	private Object data;
+
+	Operation(OperationType operationType, ArgumentType argumentType, Object data) {
```

Comment:
There is an implicit dependency between the value of `argumentType` and the type of `data`, which is not explicitly documented but just comes from how `data` is cast depending on the `argumentType`. This is hard to comprehend and error prone.

The best solution would be to use polymorphism and implement the logic in multiple classes according to the type of argument. It would be okay for me to follow a simpler solution here and simply remove the `ArgumentType` and switch over the type of `data` (and validate in the constructor that `data` is of any valid type). That is not perfect, but definitely better than the current implicit dependency between the method arguments.

Reply:
Alright, I have redesigned the Operations implementation. Please have a look again.